sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -392,6 +392,156 @@ def get_access_control(self, upn=None,  # type: Optional[bool]
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @staticmethod
+    def _set_access_control_recursive_options(mode, acl=None, **kwargs):
+        # type: (Optional[ContentSettings], Optional[Dict[str, str]], **Any) -> Dict[str, Any]
+
+        options = {
+            'mode': mode,
+            'timeout': kwargs.pop('timeout', None),
+            'continuation': kwargs.pop('continuation', None),
+            'max_records': kwargs.pop('batch_size', None),
+            'acl': acl,
+            'cls': return_headers_and_deserialized}
+        options.update(kwargs)
+        return options
+
+    def set_access_control_recursive(self,
+                                     acl=None,  # type: Optional[str]
```

Comment:
Why does `acl` have a default of None when we raise an error below if it's not set? Seems that this value should not be optional?

Reply:
Ooops! Sorry for the oversight